src/main/scala/GithubRelease.scala

```diff
@@ -20,14 +21,14 @@ case object GithubRelease {
     lazy val ghreleaseNotes         = settingKey[TagName => String]("Release notes for the given tag")
     lazy val ghreleaseTitle         = settingKey[TagName => String]("The title of the release")
     lazy val ghreleaseIsPrerelease  = settingKey[TagName => Boolean]("A function to determine release as a prerelease based on the tag name")
-
+    lazy val ghreleaseGithubToken   = settingKey[Option[String]]("Credentials for accessing the GitHub API")
     lazy val ghreleaseAssets        = taskKey[Seq[File]]("The artifact files to upload")
 
     // TODO: remove this, make them tasks or parameters for the main task
     // lazy val draft = settingKey[Boolean]("true to create a draft (unpublished) release, false to create a published one")
 
-    lazy val ghreleaseGetCredentials = taskKey[GitHub]("Checks authentification and suggests to create a new oauth token if needed")
-    lazy val ghreleaseGetRepo        = taskKey[GHRepository]("Checks repo existence and returns it if it's fine")
+    lazy val ghreleaseGetCredentials = taskKey[String]("Checks authentification and suggests to create a new oauth token if needed")
```

Comment:
This key is declared, but is not bound in the plugin. So you can either remove it completely, or (better) define as this piece of code:

https://github.com/EmilDafinov/sbt-github-release/blob/8ab82d86f8c011907804e79e1257a46d347de56b/src/main/scala/GithubRelease.scala#L80-L86

And reuse in `ghreleaseGetRepo` definition. This way user will have a way to check that credentials are set and valid.